sktime/forecasting/trend/_spline_trend_forecaster.py

```diff
@@ -0,0 +1,291 @@
+    ... )
+    >>> forecaster.fit(y)
+    SplineTrendForecaster()
+    >>> y_pred = forecaster.predict(fh=[1, 2, 3])
+    """
+
+    _tags = {
+        "authors": ["jgyasu", "tensorflow-as-tf", "mloning", "aiwalter", "fkiraly"],
+        "ignores-exogeneous-X": True,
+        "requires-fh-in-fit": False,
+        "handles-missing-data": False,
+        "capability:pred_int": True,
+    }
+
+    def __init__(
+        self,
+        regressor=None,
+        degree=1,
+        with_intercept=True,
+        prediction_intervals=False,
+        n_knots=5,
+        knots="uniform",
+        extrapolation="constant",
+    ):
+        super().__init__(
+            regressor=regressor,
+            degree=degree,
+            with_intercept=with_intercept,
+            prediction_intervals=prediction_intervals,
+        )
+        self.n_knots = n_knots
+        self.knots = knots
+        self.extrapolation = extrapolation
+        # prediction_intervals : bool, default=False
+        # By default, the extra information needed to later generate the prediction
+        # intervals is not calculated. If set to True, the extra information is
+        # calculated and stored in the forecaster.
+
+        self.set_tags(**{"capability:pred_int": prediction_intervals})
+
+    def _fit(self, y, X, fh):
+        """Fit to training data.
+
+        Parameters
+        ----------
+        y : pd.Series
+            Target time series with which to fit the forecaster.
+        X : pd.DataFrame, default=None
+            Exogenous variables are ignored
+        fh : int, list or np.array, default=None
+            The forecasters horizon with the steps ahead to to predict.
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        if self.regressor is None:
+            regressor = LinearRegression(fit_intercept=False)
+        else:
+            regressor = clone(self.regressor)
+
+        # make pipeline with spline features
+        self.regressor_ = make_pipeline(
+            SplineTransformer(
+                degree=self.degree,
+                n_knots=self.n_knots,
+                knots=self.knots,
+                extrapolation=self.extrapolation,
+                include_bias=False,
+            ),
+            regressor,
+        )
+
+        # we regress index on series values
+        # the sklearn X is obtained from the index of y
+        # the sklearn y can be taken as the y seen here
+        X_sklearn = _get_X_numpy_int_from_pandas(y.index)
+
+        # fit regressor
+        self.regressor_.fit(X_sklearn, y)
+
+        if self.prediction_intervals:
+            # calculate and save values needed for the prediction interval method
+            fitted_values = self.regressor_.predict(X_sklearn)
+            residuals = y - fitted_values
+            p = self.degree + int(self.with_intercept)
+            self.s_squared_ = np.sum(residuals**2) / (len(y) - p)
+            self.train_index_ = y.index
```

Comment:
Question to @fkiraly @benHeid : I remember there were some discussions about remembering data and/or not, do not remember details. If we store this type of info for history, do we need to check some config (store only if allowed or etc.)?

Reply:
Obsolete due to removal of prediction intervals.

More generally, this is fine - the estimator itself can decide what to remember.